src/test/java/io/confluent/kafkarest/unit/UriUtilsTest.java

```diff
@@ -79,4 +79,34 @@ public void testAbsoluteURIBuilderWithPort() throws RestConfigException {
     EasyMock.verify(uriInfo);
   }
 
+  @Test
+  public void testAbsoluteURIBuilderWithListenerForHttp() throws RestConfigException {
+    Properties props = new Properties();
+    props.put(KafkaRestConfig.HOST_NAME_CONFIG, "bar.net");
+    props.put(KafkaRestConfig.LISTENERS_CONFIG, "http://0.0.0.0:9091,https://0.0.0.0:9092");
```

Comment:
add tests for IPv6 addresses:
 * full address, no port - e.g. `fe80:0:1:2:3:4:5:6`
 * full address with port - e.g., `[fe80:0:1:2:3:4:5:6]:9092` 
 * truncated address, no port, e.g., `fe80::1`
 * truncated address, with port `[fe80::1]:9092`